internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -101,78 +103,95 @@ func (r *ClusterDeploymentsReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	clusterInstall := &hiveext.AgentClusterInstall{}
+	aciName := clusterDeployment.Spec.ClusterInstallRef.Name
```

Comment:
It's probably no required so need to check if `ClusterInstallRef` is not nil

Reply:
Done